embassy-stm32/src/timer/simple_pwm.rs

```diff
@@ -135,12 +139,90 @@ impl<'d, T: CaptureCompare16bitInstance> SimplePwm<'d, T> {
         self.inner.set_compare_value(channel, duty)
     }
 
+    /// Get the duty for a given channel.
+    ///
+    /// The value ranges from 0 for 0% duty, to [`get_max_duty`](Self::get_max_duty) for 100% duty, both included.
+    pub fn get_duty(&self, channel: Channel) -> u16 {
+        self.inner.get_compare_value(channel)
+    }
+
     /// Set the output polarity for a given channel.
     pub fn set_polarity(&mut self, channel: Channel, polarity: OutputPolarity) {
         self.inner.set_output_polarity(channel, polarity);
     }
 }
 
+impl<'d, T: CaptureCompare16bitInstance + Basic16bitInstance> SimplePwm<'d, T> {
```

Comment:
this shouldn't need a separate impl block anymore

Reply:
So, we will require `T: CaptureCompare16bitInstance + Basic16bitInstance` to use every other associate-functions&methods. I don't know whether this will add extra (maybe unneeded) restraint to those associate-functions&methods other than `.gen_waveform()`.

Reply:
CaptureCompare16bitInstance inherits from GeneralPurpose16bitInstance which inherits from Basic16bitInstance.

So `T: CaptureCompare16bitInstance + Basic16bitInstance` is the same as `T: CaptureCompare16bitInstance`

Reply:
But we still need to expicitly write `T: CaptureCompare16bitInstance + Basic16bitInstance` instead of `T: CaptureCompare16bitInstance` right?
I find if I use the less constrait one, rustc will put a error "the trait bound `T: Basic16bitInstance` is not implemented for `T`"
(Sorry I still get confused on rust trait system)

Reply:
ah, there were a few missing supertrait bounds (the lowlevel traits did inherit, the public ones didn't).  I've added them.